safe-global/safe-wallet-web
# feat: add recovery transaction validation

## What it solves

Resolves [recovery attempt validation](https://www.notion.so/safe-global/Help-is-required-with-the-next-safe-gor-0x6Eb821dA05DB250F78945F9a2a25A6CBc436010f-887942b5f46240aeb6fe1e9810496bfb)

## How this PR fixes it

The description of a proposal now checks the owner length of the proposed setup and if it is empty then an error is displayed. On top of this, static calls of all recovery interactions have been added accordingly and are displayed before execution of proposal, execution and cancellation.

## How to test it

- Propose a recovery attempt and then modify the owner setup before execution and observe the new error.
- The validation errors are otherwise hard to test without removing safety checks. The request could potentially be blocked.

## Screenshots

![image](https://github.com/safe-global/safe-wallet-web/assets/20442784/10aaef39-3870-440e-a7e1-ae7131a41e5c)

Please note, the following errors are "forced" by removing safety checks:

![image](https://github.com/safe-global/safe-wallet-web/assets/20442784/25395f7f-1ab2-48e6-9b30-874d732fa876)

![image](https://github.com/safe-global/safe-wallet-web/assets/20442784/a6cdd2a5-2c44-4dc0-86b6-8429d3728139)

## Checklist
